### PR TITLE
Switch some tests to the catch2 framework

### PR DESCRIPTION
**Pull request by m-kuhn**
It's cleaner and offers better output

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for f1451573b5f96c290eb0afd1781b2e3b5736639c:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-53117a4ee1f1ca24af756747c24e870dd501f9bd-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-53117a4ee1f1ca24af756747c24e870dd501f9bd-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-53117a4ee1f1ca24af756747c24e870dd501f9bd-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-53117a4ee1f1ca24af756747c24e870dd501f9bd-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 047c14787b0b12789548f88423aac3c4692212fa:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-634500bbdcd41cb158af5871fff369e75ea3220c-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-634500bbdcd41cb158af5871fff369e75ea3220c-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-634500bbdcd41cb158af5871fff369e75ea3220c-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-634500bbdcd41cb158af5871fff369e75ea3220c-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 94e6940ced6db4c18ef349a8826030f40f96a89e:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-79d9bdbc406ba01144f57bd6b70fba90b231ae03-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-79d9bdbc406ba01144f57bd6b70fba90b231ae03-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-79d9bdbc406ba01144f57bd6b70fba90b231ae03-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-79d9bdbc406ba01144f57bd6b70fba90b231ae03-x86.apk)

**Review by 3nids — APPROVED** [review]

**Comment by m-kuhn** [comment]
Note: the other tests are either completely different (spix) or freeze (relationreference, probably missing some qt initialisation)

**Comment by m-kuhn** [comment]
@suricactus there's a low hanging fruit on the windows path separators (`/` vs. `\`) and some other issues about clientId missing, where I'm not sure what's going on.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 88d2c4fb9201fb8df910e1ea361f60c1fe6ca14c:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-6b2fdbaa7803de1660f7b70eaf6dc4022e915d47-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-6b2fdbaa7803de1660f7b70eaf6dc4022e915d47-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-6b2fdbaa7803de1660f7b70eaf6dc4022e915d47-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-6b2fdbaa7803de1660f7b70eaf6dc4022e915d47-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 351c6c1876c6149d050f31abebf093d348102efe:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-600b7b3d090040235e87772470cf30aa4bb4a9e7-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-600b7b3d090040235e87772470cf30aa4bb4a9e7-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-600b7b3d090040235e87772470cf30aa4bb4a9e7-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-600b7b3d090040235e87772470cf30aa4bb4a9e7-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 89a76b0ac4ffba0b73527c1b313cc8afdecf247b:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-6c0ebd0f2030e3ace245663d80b54cc2ca95c80f-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-6c0ebd0f2030e3ace245663d80b54cc2ca95c80f-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-6c0ebd0f2030e3ace245663d80b54cc2ca95c80f-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-6c0ebd0f2030e3ace245663d80b54cc2ca95c80f-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 2843980523ab346a2d9412a567fd0aa4bd54cd61:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-45ce7300e7199fb9872af4ae0360deb3313c3101-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-45ce7300e7199fb9872af4ae0360deb3313c3101-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-45ce7300e7199fb9872af4ae0360deb3313c3101-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-45ce7300e7199fb9872af4ae0360deb3313c3101-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for c8f8c92535e9ad01d923f33c3c78de35d00fba5c:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-1af7ce2efa18f8cd531c67fef32bdda1196cc33d-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-1af7ce2efa18f8cd531c67fef32bdda1196cc33d-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-1af7ce2efa18f8cd531c67fef32bdda1196cc33d-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-1af7ce2efa18f8cd531c67fef32bdda1196cc33d-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 8c2cf82cb7c0faead973162c72e581978ae0f1ce:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-ee0c2c85b6b128c7137c4d86416f436f073bad05-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-ee0c2c85b6b128c7137c4d86416f436f073bad05-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-ee0c2c85b6b128c7137c4d86416f436f073bad05-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-ee0c2c85b6b128c7137c4d86416f436f073bad05-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for f88906543695af49e6b5ebb72f420d87968430f0:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-0f3effe0a5a2662af351b3b40a6fc0f116d27751-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-0f3effe0a5a2662af351b3b40a6fc0f116d27751-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-0f3effe0a5a2662af351b3b40a6fc0f116d27751-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-0f3effe0a5a2662af351b3b40a6fc0f116d27751-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for ac55f13a8f507067c5f794f790ac410898636233:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-af6650dd122d476d3ba59b77ed347323ac6ff91c-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-af6650dd122d476d3ba59b77ed347323ac6ff91c-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-af6650dd122d476d3ba59b77ed347323ac6ff91c-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-af6650dd122d476d3ba59b77ed347323ac6ff91c-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for a2d5f46426156ff5822c6dd5b6f5236189f79d3b:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-5146628862c021fc4dff5bf1130c2bf3be692127-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-5146628862c021fc4dff5bf1130c2bf3be692127-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-5146628862c021fc4dff5bf1130c2bf3be692127-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-5146628862c021fc4dff5bf1130c2bf3be692127-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 914fd38f676ca54bf416aac8ff4b7e6957331380:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-a83bf65fc678bfcf47c9a436a75539778d3a9194-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-a83bf65fc678bfcf47c9a436a75539778d3a9194-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-a83bf65fc678bfcf47c9a436a75539778d3a9194-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-a83bf65fc678bfcf47c9a436a75539778d3a9194-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for f65b7be833f8d27ce6fb05235d259fa6b76d8958:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-b75baf31a0cdeecafc122aedf5db88f713d43504-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-b75baf31a0cdeecafc122aedf5db88f713d43504-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-b75baf31a0cdeecafc122aedf5db88f713d43504-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-b75baf31a0cdeecafc122aedf5db88f713d43504-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 3c8926b518074985ce054d5430a6cd04a53fd735:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-8ebc19ee78649fc1e18b851344238f937941bbe2-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-8ebc19ee78649fc1e18b851344238f937941bbe2-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-8ebc19ee78649fc1e18b851344238f937941bbe2-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-8ebc19ee78649fc1e18b851344238f937941bbe2-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 43831a4f4a3342c9006c07e485486ce933f13946:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-e3faa62a0ecdcfc367ae2ca6aecf09f5dc8deba0-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-e3faa62a0ecdcfc367ae2ca6aecf09f5dc8deba0-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-e3faa62a0ecdcfc367ae2ca6aecf09f5dc8deba0-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-e3faa62a0ecdcfc367ae2ca6aecf09f5dc8deba0-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 67785e23f4597b9e3b5d97f1570ae7a66775e7a1:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-726cacea21ddd1da3cd0c24ddf588b9a6cef182f-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-726cacea21ddd1da3cd0c24ddf588b9a6cef182f-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-726cacea21ddd1da3cd0c24ddf588b9a6cef182f-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-726cacea21ddd1da3cd0c24ddf588b9a6cef182f-x86.apk)

File: src/core/utils/qfieldcloudutils.cpp [hunk 1]
```diff
@@ -63,12 +67,14 @@ const QString QFieldCloudUtils::getProjectId( const QString &fileName )
 
 void QFieldCloudUtils::setProjectSetting( const QString &projectId, const QString &setting, const QVariant &value )
 {
+  thread_local QgsSettings settings;
   const QString projectPrefix = QStringLiteral( "QFieldCloud/projects/%1" ).arg( projectId );
-  return QSettings().setValue( QStringLiteral( "%1/%2" ).arg( projectPrefix, setting ), value );
+  settings.setValue( QStringLiteral( "%1/%2" ).arg( projectPrefix, setting ), value );
```
Comment (m-kuhn) [inline src/core/utils/qfieldcloudutils.cpp:72]: @suricactus did you have a reason to use QSettings() instead of QgsSettings()?

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for eb387d3a389ebcfc58488ed55928f42c129ed617:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-e8750d30c1275f75e87cd328ef675f842be8c923-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-e8750d30c1275f75e87cd328ef675f842be8c923-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-e8750d30c1275f75e87cd328ef675f842be8c923-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-2081-e8750d30c1275f75e87cd328ef675f842be8c923-x86.apk)

File: src/core/utils/qfieldcloudutils.cpp (hunk 1 above)
Comment (suricactus) [reply src/core/utils/qfieldcloudutils.cpp:72]: No particular reason, I guess I was unaware at the time of writing.

**Review by suricactus — APPROVED** [review]

**Pull request merged**